src/main/java/org/alliancegenome/curation_api/services/mati/MaTIService.java

```diff
@@ -0,0 +1,79 @@
+package org.alliancegenome.curation_api.services.mati;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import org.alliancegenome.curation_api.model.entities.IdentifiersRange;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.io.IOException;
+import java.util.Base64;
+
+@ApplicationScoped
+public class MaTIService {
+//	@ConfigProperty(name = "okta.client.id") TODO: reading @ConfigProperties is broken inside integ tests
+	private String client_id;
+//	@ConfigProperty(name = "okta.client.secret") TODO: workaround with env vars needs github actions support
+	private  String client_secret;
+//	@ConfigProperty(name = "okta.url")
+	private String okta_url;
+//	@ConfigProperty(name = "okta.scopes")
+	private String okta_scopes;
+//	@ConfigProperty(name = "mati.url")
+	private String mati_url;
+
+	public MaTIService() {
+		client_id = System.getenv("OKTA_CLIENT_ID");
+		client_secret = System.getenv("OKTA_CLIENT_SECRET");
+		okta_url = System.getenv("OKTA_URL");
+		okta_scopes = System.getenv("OKTA_SCOPES");
+		mati_url = System.getenv("MATI_URL");
+	}
+
+	private String fetchOktaToken() throws IOException {
+		String authorization = "Basic " +
+			Base64.getEncoder().encodeToString((client_id+":"+client_secret).getBytes());
+		String token = RestAssured.given().
+			contentType(ContentType.URLENC).
+			header("Accept", "application/json").
+			header("Cache-Control", "no-cache").
+			header("Authorization", authorization).
+			formParam("grant_type", "client_credentials").
+			formParam("scope", okta_scopes).
+			when().
+			post(okta_url + "/oauth2/default/v1/token").
+			then().
+			extract().path("access_token").toString();
+		return token;
```

Comment:
And using the RestProxy via the library will prevent all the hardcoding of these headers and credentials... etc.

Reply:
sorry what I mean by rest proxy is:
```java
private IdentifierResourceRESTInterface mati_api = RestProxyFactory.createProxy(IdentifierResourceRESTInterface.class, mati_base_url);
Response resp = mati_api.increment(authorization, subdomain, value)
```
Plus the integration tests are not going to work via github actions because MATI is behind a firewall. So I would think so long as they might work locally, then we can disable them when pushing the code.

Reply:
The tests run locally with ENV var

Reply:
They will also run via @ConfigProperty we do that for all our other tests.

Reply:
The argument authorization would need a token. So, the use of RestProxyFactory would still need to add more code.